### PR TITLE
feat: Allow max value for progress bar + add 'ratio' type

### PR DESCRIPTION
**Pull request by akessaris**
### Description

Allow progress bar to show ratio ("numerator / denominator") rather than just percentage in order to more accurately understand the progress scope.

Issue AWSUI-20321

### How has this been tested?

Unit + integration tests. Added to permutations for visual regression tests.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/progress-bar/interfaces.ts [hunk 1]
```diff
@@ -72,4 +77,5 @@ export interface ProgressBarProps extends BaseComponentProps {
 export namespace ProgressBarProps {
   export type Status = 'in-progress' | 'success' | 'error';
   export type Variant = 'standalone' | 'flash' | 'key-value';
+  export type ContentType = 'percentage' | 'ratio';
```
Comment (akessaris) [inline src/progress-bar/interfaces.ts:80]: Unrelated change, will remove in next update

**Comment by akessaris** [comment]
I didn't update existing test utils or css classes that explicitly mention "percentage", but happy to update for consistency.

File: src/progress-bar/interfaces.ts [hunk 2]
```diff
@@ -69,7 +69,43 @@ export interface ProgressBarProps extends BaseComponentProps {
...
+  /**
+   * Control localization of the progress bar value.
+   *
+   * @defaultValue If type === `percentage`, `${value}%`
+   * @defaultValue If type === `ratio`, `${value} of ${maxValue}`
+   */
+  ariaValueText?: string;
+}
+
+export type ProgressBarProps = PercentageProgressBarProps | RatioProgressBarProps;
```
Comment (timogasda) [inline src/progress-bar/interfaces.ts:105]: I'm not sure if our documenter and typedoc parser can handle this. There's some expectations around how public components define their interfaces, so that we can automatically generate the API docs for the website.

I would suggest we play it safe and use the regular exported interface for both progress bar types

File: src/progress-bar/interfaces.ts [hunk 3]
```diff
@@ -69,7 +69,43 @@ export interface ProgressBarProps extends BaseComponentProps {
   onResultButtonClick?: NonCancelableEventHandler;
 }
 
+interface PercentageProgressBarProps extends BaseProgressBarProps {
+  /**
+   * Specifies the progress type.
+   *
+   * @defaultValue 'percentage'
```
Comment (timogasda) [inline src/progress-bar/interfaces.ts:76]: Nice idea, but we parse public interface comments to generate our public API documentation. Please keep these comments limited to documentation strings.

File: src/progress-bar/internal.tsx [hunk 4]
```diff
@@ -11,39 +11,77 @@ import InternalStatusIndicator from '../status-indicator/internal';
...
-  const progressValue = clamp(roundedValue, 0, MAX_VALUE);
+  const progressValue = clamp(roundedValue, 0, maxValue);
+  const isRatio = type === 'ratio';
+  const percentage = isRatio && maxValue !== 100 ? Math.round(((progressValue * 1.0) / maxValue) * 100) : progressValue;
+  const valueText = isRatio ? ariaValueText || `${progressValue}/${maxValue}` : `${percentage}%`;
+
+  const [assertion, setAssertion] = useState('');
+  const throttledAssertion = useMemo(() => {
+    return throttle((value: ProgressBarProps['value']) => {
+      const announcement = type === 'ratio' ? `${value} of ${maxValue}}` : `${value}%`;
```
Comment (timogasda) [inline src/progress-bar/internal.tsx:51]: Shouldn't the announcement be `valueText`?

File: src/progress-bar/internal.tsx [hunk 5]
```diff
@@ -11,39 +11,77 @@ import InternalStatusIndicator from '../status-indicator/internal';
...
+        <progress
+          className={clsx(
+            styles.progress,
+            progressValue >= maxValue && styles.complete,
+            isInFlash && styles['progress-in-flash']
+          )}
+          max={maxValue}
+          value={progressValue}
+          aria-valuemin={0}
+          aria-valuemax={maxValue}
```
Comment (timogasda) [inline src/progress-bar/internal.tsx:72]: We've removed these in the past for the regular percentage type (https://github.com/cloudscape-design/components/pull/1082). I would suggest that we only add the new aria props for the new ratio type, and keep the old percentage props unchanged.

But also, let's first check if `aria-valuemin`, `aria-valuemax`, and `aria-valuenow` are even needed for the ratio type. Maybe `min`, `max` and `value` are sufficient here as well.
